### DISCUSSION TITLE
BaseEdge style attribute zIndex does not take effect

### DISCUSSION THREAD

**Discussion by egglemmon** [bcakmakoglu/vue-flow#1691] [Q&A] [answered]
When using BaseEdge, zIndex setting does not take effect
![zIndex](https://github.com/user-attachments/assets/37ce2346-95e5-4841-9ac6-cd3067acd79e)

**Comment by bcakmakoglu** [comment]
That is expected because the BaseEdge does *not* handle z-indexing. It is handled by the wrapping element `.vue-flow__edges`
If you need to change the z-index of an edge use `edge.zIndex`.

**Comment by egglemmon** [comment]
If used in a nested node, the mouse will not be able to select the edge. This will result in the inability to delete the edge. How should zIndex be set when there are many child nodes in the nested node.
I observed that when using the default edge, zIndex will automatically set a larger value.

**Reply by bcakmakoglu** [reply, marked as answer]
Just use `elvateEdgesOnSelect` and it should automatically handle this for you.

**Comment by egglemmon** [comment]
How can I make the parent node adaptively resize after the nested node expands the parent node? It only seems to expand infinitely.

**Reply by egglemmon** [reply]
Is using: UpdateNodeDimensions() ? If so, I don't know how to get the parameter nodeElement

**Reply by bcakmakoglu** [reply]
What? ^^

**Reply by egglemmon** [reply]
![image](https://github.com/user-attachments/assets/5fbceaf7-89fb-4dcc-9463-0c62b42076cc)
That’s how to narrow the scope of the parent node of the nested node to the red box

**Reply by egglemmon** [reply]
After expansion, when the child node moves to the left, a large area is left on the right side of the parent node. How to make the parent node shrink based on the movement of the child node (eg: red box)

**Comment by egglemmon** [comment]
Using addNodes() adds new child nodes within a nested node, but the parent node does not resize automatically. How to automatically resize parent node? But after dragging this child node, the parent node can automatically resize. How to trigger automatic resizing?

**Reply by bcakmakoglu** [reply]
You'll have to calculate that yourself.
Expand parent is not meant for programmatic resizing, it's only meant for resizing with a drag action.
It's a limited feature.
